src/js/ui/simply-pebble.js

```diff
@@ -140,7 +207,43 @@ var colorMap = {
 };
 
 var Color = function(color) {
-  return colorMap[color] ? colorMap[color] : colorMap.clear;
+  if (color.charAt(0)=='#') {
+      //Convert full hex to shorthand
+      if (color.length==4){
+          var r = color.charAt(1);
+          var g = color.charAt(2);
+          var b = color.charAt(3);
+          color = '#'+r+r+g+g+b+b;
+      }
+      //Ensure upper case
+      color = color.toUpperCase();
+      return hexColorMap[roundColor(color)];
+  }
+  return namedColorMap[color] ? namedColorMap[color] : namedColorMap.clear;
+};
+
+var roundColor = function (color) {
+    var closestTo = function(color, colors) {
+        var dist = Infinity;
+        var result = color;
+        colors.forEach(function(col) {
+            var localdist = Math.abs(parseInt(color, 16)-parseInt(col, 16));
```

Comment:
`localdist` should be `localDist`, and there should be spaces between the `-`. Alternatively, you can rename the first `dist` to `nearestDist` and rename `localdist` to just `dist`.